### PR TITLE
Migrate continuous integration to GitHub actions (fixes #652)

### LINKED ISSUE #652: Travis CI free usage ends Dec 3; mozilla repos should switch to other CI platforms
We're opening this issue because your project has used Travis CI within the last 6 months. If you have already migrated off it, you can close and ignore this issue.

Travis CI is ending free builds on public repositories. travis-ci.com [stopped providing](https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing)them in early November, and travis-ci.org [will stop](https://mailchi.mp/3d439eeb1098/travis-ciorg-is-moving-to-travis-cicom) after December 31, 2020. To avoid disruptions to your workflows, you must migrate to another CI service.

For production use cases, we recommend switching to [CircleCI](https://circleci.com/). This service is already widely used within Mozilla. There is a guide to migrating from Travis CI to CircleCI available [here](https://circleci.com/docs/2.0/migrating-from-travis/).

For non production use cases, we recommend either CircleCI or [Github Actions](https://github.com/features/actions). There is a guide to migrating from Travis CI to Github Actions available [here](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/migrating-from-travis-ci-to-github-actions). Github Actions usage within Mozilla is new, and you will have to work with our github administrators to enable specific actions following [this process](https://wiki.mozilla.org/GitHub#How_do_I_hook_up_a_new_GitHub_Actions_or_3rd_party_application_to_a_repository_in_the_mozilla_org.3F).

If you have any questions, reach out in `#github-admin:mozilla.org` on [matrix](https://chat.mozilla.org/#/room/#github-admin:mozilla.org).

### PR DESCRIPTION
**Pull request by wlach**
This is more urgent now that [travis ci is shutting down for real](https://blog.travis-ci.com/2021-05-07-orgshutdown)

## Description

Describe what this change does.

- [x] (N/A) I have documented this change in the design system.
- [x] I have recorded this change in `CHANGELOG.md`.

### Issue

This fixes #652 

### Testing

N/A, check ci status

### CODE REVIEW

**Comment by wlach** [comment]
Seems to work on my fork: https://github.com/wlach/protocol/actions/runs/854424933

**Comment by craigcook** [comment]
This looks good to me and works in my testing, but I'm outside my wheelhouse.
@pmac Could you give this a look when you've got a minute, make sure we're not missing something?

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    branches: [ main ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [10.x, 12.x, 14.x, 15.x]
```
Comment (craigcook) [inline .github/workflows/node.js.yml:16]: Our minimum Node version will bump up to 12.x in https://github.com/mozilla/protocol/pull/691, so we can drop 10.x from this list.

**Review by craigcook — CHANGES_REQUESTED** [review]

File: .github/workflows/node.js.yml (hunk 1 above)
Comment (wlach) [inline .github/workflows/node.js.yml:16]:
```suggestion
        node-version: [12.x, 14.x, 15.x]
```

**Review by craigcook — APPROVED** [review]

**Pull request merged**